### PR TITLE
update changelog.d format and Matrix link

### PR DESCRIPTION
**Pull request by geekosaur**
`changelog-d` was changed to make `packages` a list, so reflect this in the example. Also remove the "omit if it's an overarching change" part: we must list all packages if we want it to show up when generating release changelogs.

While I was there, I updated the Matrix channel reference because it was still reflecting the EMS Libera bridge.

**Template B: This PR does not modify behaviour or interface**

*E.g. the PR only touches documentation or tests, does refactorings, etc.*

Include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [ ] Is this a PR that fixes CI? If so, it will need to be backported to older cabal release branches (ask maintainers for directions).

### CODE REVIEW

**Review by 9999years — APPROVED** [review]

**Review by ulysses4ever — APPROVED** [review]
It's sad that changelog-d chose to use square brackets. Just a comma-separated list would be the least surprise.

**Comment by geekosaur** [comment]
It'd still have been a syntax change: originally it was space-separated.

**Comment by fgaz** [comment]
> `changelog-d` was changed to make `packages` a [bracketed] list

...I don't think this ever happened?!

```sh
cd "$(mktemp -d)"

cat > entry1 <<EOF
packages: foo
synopsis: entry 1
EOF

cat > entry2 <<EOF
packages: [foo]
synopsis: entry 2
EOF

cat > config <<EOF
organization: org
repository: repo
EOF
```

```console
$ changelog-d-v1.0.1-x86_64-linux . --package foo
Errors encountered when parsing file ./entry2:

entry2:1:11: error:
unexpected '['
expecting white space or end of input

    1 | packages: [foo]
      |           ^
$ rm entry2
$ changelog-d-v1.0.1-x86_64-linux . --package foo
- entry 1
```

**Comment by geekosaur** [comment]
https://github.com/haskell/cabal/actions/runs/11061350227/job/30733804747

**Comment by fgaz** [comment]
I see what happened. That CI job ran against 49d2ab07cb6afba8d31e663500228c3dda9de711, which includes a [changelog entry in markdown format](https://github.com/haskell/cabal/blob/49d2ab07cb6afba8d31e663500228c3dda9de711/changelog.d/pr-10273). The markdown format uses a YAML frontmatter, and YAML lists do require brackets when using the inline syntax. This was introduced in [changelog-d#5](https://codeberg.org/fgaz/changelog-d/pulls/5) and does not affect cabal-formatted entries.

The markdown example in the changelog-d readme was slightly incorrect, I [just fixed it](https://codeberg.org/fgaz/changelog-d/commit/0120b619ab3e4827914234674c0a3c7009642901).

In the future I may allow space-separated packages [as it's done for issues and prs](https://codeberg.org/fgaz/changelog-d/src/commit/17365ec17d89c875d3b4b0c4ce6295c31e72acb2/src/ChangelogD.hs#L358), though this is a breaking change.

**Comment by geekosaur** [comment]
Documentation corrected and a note about Markdown format added.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -344,14 +344,15 @@ description: {
 
 }
 ```
+(Note: if you choose to use `changelog-d`'s Markdown format, `packages` must be a bracketed list.)
```
Comment (ulysses4ever) [inline CONTRIBUTING.md:347]: Is it comma-separated? (E.g. nix uses brackets but no commas)

Comment (ulysses4ever) [reply CONTRIBUTING.md:347]: I also suggest removing the parenthesis.

Comment (ulysses4ever) [reply CONTRIBUTING.md:347]: I also realized that this phrase mak s me thing that the "old format" doesn't support markdown at all, which is false. It seems like more with needs to be done with the explanation. Maybe referencing the changelog-d own doc would be more appropriate. But in any case I find the current version confusing.

Comment (geekosaur) [reply CONTRIBUTING.md:347]: I wanted to link the `README.md` but the description of "markdown-frontmatter" format isn't a separate link.

Comment (geekosaur) [reply CONTRIBUTING.md:347]: It's brackets and commas.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -345,13 +345,32 @@ description: {
...
+prs: #0000
+issues: #0000 #0000
+significance: significant
+---
+
+- Detail number 1
+- Detail number 2
+
+```
+The package list must be enclosed in square brackets, but they aren't needed for `prs` or `issues`.
```
Comment (ulysses4ever) [inline CONTRIBUTING.md:365]:
```suggestion
The package list must be enclosed in square brackets and comma-separated, but this isn't needed for `prs` or `issues`.
```

**Review by ulysses4ever — APPROVED** [review]
I still think comma-separated should be explicitly mentioned.

File: CONTRIBUTING.md (hunk 2 above)
Comment (geekosaur) [reply CONTRIBUTING.md]: Sigh. I could have sworn I had added that; possibly I inadvertently removed it when I re-rewrote the markdown-frontmatter part.

Comment (ulysses4ever) [reply CONTRIBUTING.md]: No worries!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -348,13 +348,32 @@ description: {
...
+Changelogs may also be written in "markdown-frontmatter" format. This is useful if your
+description contains braces, which must be escaped with backslashes in `.cabal` file
+format.
+
+```markdown
+---
+synopsis: Add feature xyz
+packages: [cabal-install]
+prs: #0000
+issues: #0000 #0000
```
Comment (9999years) [inline CONTRIBUTING.md:360]: Oh, this needs a followup: in YAML, `#` is a comment marker, so the changelog-d YAML format doesn't use the `#` in issue/PR numbers.

Comment (geekosaur) [reply CONTRIBUTING.md:360]: I copied that from `changelog-d`'s `README`; I guess that was what [this](https://github.com/haskell/cabal/pull/10383#issuecomment-2378858091) was talking about.

**Pull request merged**